src/frontend/src/optimizer/plan_node/stream_global_approx_percentile.rs

```diff
@@ -95,8 +97,42 @@ impl PlanTreeNodeUnary for StreamGlobalApproxPercentile {
 impl_plan_tree_node_for_unary! {StreamGlobalApproxPercentile}
 
 impl StreamNode for StreamGlobalApproxPercentile {
-    fn to_stream_prost_body(&self, _state: &mut BuildFragmentGraphState) -> PbNodeBody {
-        todo!()
+    fn to_stream_prost_body(&self, state: &mut BuildFragmentGraphState) -> PbNodeBody {
+        let relative_error = self.relative_error.get_data().as_ref().unwrap();
+        let relative_error = relative_error.as_float64().into_inner();
+        let base = (1.0 + relative_error) / (1.0 - relative_error);
+        let quantile = self.quantile.get_data().as_ref().unwrap();
+        let quantile = quantile.as_float64().into_inner();
+
+        // setup table: bucket_id->count
+        let mut bucket_table_builder = TableCatalogBuilder::default();
+        bucket_table_builder.add_column(&Field::with_name(DataType::Int16, "sign"));
+        bucket_table_builder.add_column(&Field::with_name(DataType::Int32, "bucket_id"));
+        bucket_table_builder.add_column(&Field::with_name(DataType::Int64, "count"));
+        bucket_table_builder.add_order_column(0, OrderType::ascending()); // bucket_id
+        bucket_table_builder.add_order_column(1, OrderType::ascending()); // sign
+
+        // setup table: total_count
+        let mut count_table_builder = TableCatalogBuilder::default();
+        count_table_builder.add_column(&Field::with_name(DataType::Int64, "total_count"));
```

Comment:
Is global approx percentile supposed to handle only the queries without `group by` and have singleton distribution?

Is it possible to use a special `bucket_id` for the total count so we can merge it with `bucket_table`?

Reply:
> Is global approx percentile supposed to handle only the queries without group by and have singleton distribution?

Yes.

> Is it possible to use a special bucket_id for the total count so we can merge it with bucket_table?

Possible but it will make the encoding / decoding more complex. If we use bucket_id::MAX / MIN we also have to tell our users that it's unavailable for the approx percentile.
I prefer to just use a separate state table as a result.

Reply:
Ah actually, we can use `sign` column for this purpose. Let `sign` "2" mean that it is the total row count, and we can save a state table I suppose.

I'm going to make a note of it as an optimization we can use if needed. But I won't implement it in this PR, as it makes the state more complex.

Reply:
Tracked: https://github.com/risingwavelabs/risingwave/issues/17531